Description: Fix build of documentation to build everything
 using docbook-xsl, xalan2 and xhtmlrenderer from Debian
Author: Miguel Landaeta <miguel@miguel.cc>
Author: Damien Raude-Morvan <drazzib@debian.org>
Forwarded: not-needed
Last-Update: 2012-08-18

--- a/buildSrc/src/main/groovy/org/gradle/build/docs/Docbook2XHtml.groovy
+++ b/buildSrc/src/main/groovy/org/gradle/build/docs/Docbook2XHtml.groovy
@@ -71,8 +71,8 @@
                 jvmArgs '-Xmx256m'
                 classpath ClasspathUtil.getClasspathForClass(XslTransformer)
                 classpath this.classpath
-                classpath new File(stylesheetsDir, 'extensions/xalan27.jar')
-                systemProperty 'xslthl.config', new File("$stylesheetsDir/highlighting/xslthl-config.xml").toURI()
+                classpath new File('/usr/share/java/xalan2.jar')
+                systemProperty 'xslthl.config', new File("/usr/share/xml/docbook/stylesheet/docbook-xsl/highlighting/xslthl-config.xml").toURI()
                 systemProperty 'org.apache.xerces.xni.parser.XMLParserConfiguration', 'org.apache.xerces.parsers.XIncludeParserConfiguration'
             }
         }
--- a/subprojects/docs/docs.gradle
+++ b/subprojects/docs/docs.gradle
@@ -40,23 +40,23 @@
 }
 
 configurations {
-    userGuideStyleSheets
+    //userGuideStyleSheets
     userGuideTask
     jquery
-    fonts
+    //fonts
 }
 
 dependencies {
-    userGuideTask 'xalan:xalan:2.7.1', 'xerces:xercesImpl:2.9.1'
-    userGuideTask module('xhtmlrenderer:xhtmlrenderer:R8rc1') {
+    userGuideTask 'xalan:xalan2:2.7.1', 'xerces:xercesImpl:2.9.1'
+    userGuideTask module('xhtmlrenderer:core-renderer:R8rc1') {
         dependency 'itext:itext:2.0.8@jar'
     }
     userGuideTask 'xslthl:xslthl:2.0.1@jar'
 
-    userGuideStyleSheets 'docbook:docbook-xsl:1.75.2@zip'
+    //userGuideStyleSheets 'docbook:docbook-xsl:1.75.2@zip'
     jquery "jquery:jquery.min:1.8.0@js"
 
-    fonts \
+    /*fonts \
         "lato:regular:6:v0SdcGFAl2aezM9Vq_aFTQ@ttf",
         "lato:regular-italic:6:LqowQDslGv4DmUBAfWa2Vw@ttf",
         "lato:bold:6:DvlFBScY1r-FMtZSYIYoYw@ttf",
@@ -64,7 +64,7 @@
         "ubuntumono:regular:3:ViZhet7Ak-LRXZMXzuAfkZ0EAVxt0G0biEntp43Qt6E@ttf",
         "ubuntumono:regular-italic:3:KAKuHXAHZOeECOWAHsRKA-LrC4Du4e_yfTJ8Ol60xk0@ttf",
         "ubuntumono:bold:3:ceqTZGKHipo8pJj4molytp_TkvowlIOtbR7ePgFOpF4@ttf",
-        "ubuntumono:bold-italic:3:n_d8tv_JOIiYyMXR4eaV9WsGzsqhEorxQDpu60nfWEc@ttf"
+        "ubuntumono:bold-italic:3:n_d8tv_JOIiYyMXR4eaV9WsGzsqhEorxQDpu60nfWEc@ttf"*/
 }
 
 ext {
@@ -91,7 +91,7 @@
 }
 
 tasks.withType(Docbook2Xhtml) {
-    dependsOn userguideStyleSheets
+    //dependsOn userguideStyleSheets
     classpath = configurations.userGuideTask
     stylesheetsDir = userguideStyleSheets.destinationDir
 }
@@ -121,12 +121,12 @@
         [it.name, it.bytes.encodeBase64().toString()]
     }
 
-    def fonts = configurations.fonts.resolvedConfiguration.resolvedArtifacts.collectEntries {
+    /*def fonts = configurations.fonts.resolvedConfiguration.resolvedArtifacts.collectEntries {
         def id = it.moduleVersion.id
         ["${id.group}-${id.name}".toString(), it.file.bytes.encodeBase64().toString()]
-    }
+    }*/
 
-    ext.tokens = images + fonts
+    ext.tokens = images // + fonts
     css.inputs.property 'tokens', tokens
     css.filter org.apache.tools.ant.filters.ReplaceTokens, tokens: tokens
 }
@@ -166,10 +166,10 @@
         include '*.xsl'
     }
     from(cssFiles)
-    from({ zipTree(configurations.userGuideStyleSheets.singleFile) }) {
-        // Remove the prefix
-        eachFile { fcd -> fcd.path = fcd.path.replaceFirst('^docbook-xsl-[0-9\\.]+/', '') }
-    }
+    //from({ zipTree(configurations.userGuideStyleSheets.singleFile) }) {
+    //    // Remove the prefix
+    //    eachFile { fcd -> fcd.path = fcd.path.replaceFirst('^docbook-xsl-[0-9\\.]+/', '') }
+    //}
 }
 
 task samplesDocbook(type: AssembleSamplesDocTask) {
--- a/subprojects/docs/src/docs/stylesheets/dslHtml.xsl
+++ b/subprojects/docs/src/docs/stylesheets/dslHtml.xsl
@@ -14,7 +14,7 @@
   ~ limitations under the License.
   -->
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">
-    <xsl:import href="html/chunkfast.xsl"/>
+    <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/html/chunkfast.xsl"/>
     <xsl:import href="userGuideHtmlCommon.xsl"/>
 
     <xsl:output method="html" doctype-system="http://www.w3.org/TR/html4/strict.dtd"
@@ -209,4 +209,4 @@
         <dt><xsl:apply-templates select="$segtitles[$segnum=position()]" mode="segtitle-in-seg"/>:</dt>
         <dd><xsl:apply-templates/></dd>
     </xsl:template>
-</xsl:stylesheet>
\ No newline at end of file
+</xsl:stylesheet>
--- a/subprojects/docs/src/docs/stylesheets/standaloneHtml.xsl
+++ b/subprojects/docs/src/docs/stylesheets/standaloneHtml.xsl
@@ -17,7 +17,7 @@
         xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
         xmlns:xi="http://www.w3.org/2001/XInclude"
         version="1.0">
-    <xsl:import href="html/docbook.xsl"/>
+    <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/html/docbook.xsl"/>
     <xsl:import href="userGuideHtmlCommon.xsl"/>
 
     <xsl:param name="section.autolabel">0</xsl:param>
@@ -36,4 +36,4 @@
             <xi:include href="userguide.css" parse="text"/>
         </style>
     </xsl:template>
-</xsl:stylesheet>
\ No newline at end of file
+</xsl:stylesheet>
--- a/subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl
+++ b/subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl
@@ -15,7 +15,7 @@
   -->
 <xsl:stylesheet
         xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
-    <xsl:import href="html/chunkfast.xsl"/>
+    <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/html/chunkfast.xsl"/>
     <xsl:import href="userGuideHtmlCommon.xsl"/>
 
     <xsl:param name="root.filename">userguide</xsl:param>
@@ -84,4 +84,4 @@
         </div>
     </xsl:template>
 
-</xsl:stylesheet>
\ No newline at end of file
+</xsl:stylesheet>
--- a/subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl
+++ b/subprojects/docs/src/docs/stylesheets/userGuideHtmlCommon.xsl
@@ -16,8 +16,8 @@
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                 xmlns:xslthl="http://xslthl.sf.net"
                 version="1.0">
-    <xsl:import href="highlighting/common.xsl"/>
-    <xsl:import href="html/highlight.xsl"/>
+    <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/highlighting/common.xsl"/>
+    <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/html/highlight.xsl"/>
 
     <xsl:output method="html"
                 encoding="UTF-8"
--- a/subprojects/docs/src/docs/stylesheets/userGuidePdf.xsl
+++ b/subprojects/docs/src/docs/stylesheets/userGuidePdf.xsl
@@ -14,7 +14,7 @@
   ~ limitations under the License.
   -->
 <xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
-    <xsl:import href="xhtml/docbook.xsl"/>
+    <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/xhtml/docbook.xsl"/>
     <xsl:import href="userGuideHtmlCommon.xsl"/>
 
     <xsl:output method="xml"
@@ -59,4 +59,4 @@
             <xsl:apply-templates select=".." mode="title.markup"/>
         </h1>
     </xsl:template>
-</xsl:stylesheet>
\ No newline at end of file
+</xsl:stylesheet>
--- a/subprojects/docs/src/docs/stylesheets/userGuideSingleHtml.xsl
+++ b/subprojects/docs/src/docs/stylesheets/userGuideSingleHtml.xsl
@@ -15,6 +15,6 @@
   -->
 <xsl:stylesheet
         xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
-    <xsl:import href="html/docbook.xsl"/>
+    <xsl:import href="/usr/share/xml/docbook/stylesheet/docbook-xsl/html/docbook.xsl"/>
     <xsl:import href="userGuideHtmlCommon.xsl"/>
-</xsl:stylesheet>
\ No newline at end of file
+</xsl:stylesheet>
